code/modules/reagents/reagents/_reagents.dm

```diff
@@ -186,13 +203,136 @@
 	return
 
 /datum/reagent/proc/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+	if(nutriment_factor)
+		if(!injectable && alien != IS_SLIME)
+			M.adjustToxLoss(0.1 * removed)
+			return
+		affect_ingest(M, alien, removed)
+
+	if(tox_strength)
+		tox_strength *= M.species.chem_strength_tox
+		if(tox_strength && alien != IS_DIONA)
+			if(issmall(M)) removed *= 2 // Small bodymass, more effect from lower volume.
+			if(alien == IS_SLIME)
+				removed *= 0.25 // Results in half the standard tox as normal. Prometheans are 'Small' for flaps.
+				if(dose >= 10)
+					M.adjust_nutrition(tox_strength * removed) // Body has to deal with the massive influx of toxins, rather than try using them to repair.
+				else
+					M.heal_organ_damage((10/tox_strength) * removed, (10/tox_strength) * removed) //Doses of toxins below 10 units, and 10 tox_strength, are capable of providing useful compounds for repair.
+			M.adjustToxLoss(tox_strength * removed)
+		
+
+	if(booze_strength)
+		if(issmall(M)) removed *= 2
+		var/strength_mod = 3 * M.species.alcohol_mod //Alcohol is 3x stronger when injected into the veins.
+		if(alien == IS_SKRELL)
+			strength_mod *= 5
+		if(alien == IS_TAJARA)
+			strength_mod *= 1.25
+		if(alien == IS_UNATHI)
+			strength_mod *= 0.75
+		if(alien == IS_DIONA)
+			strength_mod = 0
+		if(alien == IS_SLIME)
+			M.adjustToxLoss(removed) //Sterilizing, if only by a little bit. Also already doubled above.
+
+		M.add_chemical_effect(CE_ALCOHOL, 1)
+		var/effective_dose = dose * strength_mod * (1 + volume/60) //drinking a LOT will make you go down faster
+
+		if(effective_dose >= booze_strength) // Early warning
+			M.make_dizzy(18) // It is decreased at the speed of 3 per tick
+		if(effective_dose >= booze_strength * 2) // Slurring
+			M.slurring = max(M.slurring, 90)
+		if(effective_dose >= booze_strength * 3) // Confusion - walking in random directions
+			M.Confuse(60)
+		if(effective_dose >= booze_strength * 4) // Blurry vision
+			M.eye_blurry = max(M.eye_blurry, 30)
+		if(effective_dose >= booze_strength * 5) // Drowsyness - periodically falling asleep
+			M.drowsyness = max(M.drowsyness, 60)
+		if(effective_dose >= booze_strength * 6) // Toxic dose
+			M.add_chemical_effect(CE_ALCOHOL_TOXIC, toxicity*3)
+		if(effective_dose >= booze_strength * 7) // Pass out
+			M.Paralyse(60)
+			M.Sleeping(90)
+
+		if(druggy != 0)
+			M.druggy = max(M.druggy, druggy*3)
+
+		if(adj_temp > 0 && M.bodytemperature < targ_temp) // 310 is the normal bodytemp. 310.055
+			M.bodytemperature = min(targ_temp, M.bodytemperature + (adj_temp * TEMPERATURE_DAMAGE_COEFFICIENT))
+		if(adj_temp < 0 && M.bodytemperature > targ_temp)
+			M.bodytemperature = min(targ_temp, M.bodytemperature - (adj_temp * TEMPERATURE_DAMAGE_COEFFICIENT))
+
+		if(halluci)
+			M.hallucination = max(M.hallucination, halluci*3)
+
+
+
 	return
 
 /datum/reagent/proc/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
+	if(nutriment_factor)
```

Comment:
Huge amount of duplicate code here. Should really move it into a proc or just have affect_blood call affect_ingest.

Reply:
Well i am just moving things around. A proper brush through to see what the deal is with those variables is a bit outside the scope of this PR

Reply:
I'm not talking about the variables - you have added identical code to affect_ingest and affect_blood that could be moved into a single proc and called, reducing the amount of code you're adding/that has to be maintained.

Reply:
I don't know how to say it politely but i really do not feel like putting 3/4 hours into making a proper, unified proc to handle this. 
And yes, while there is duplicate code, this is not an addition, given it was all pretty much here before, if only scattered across multiple files.

Reply:
It's nearly identical code. It should take maybe ten minutes to split it out.
Someone else will need tonreview/approve if you don't want to do that as this level of copypaste is something that contributes to poor long term code quality and maintainability.